product/views.py

```diff
@@ -0,0 +1,131 @@
+
+            return JsonResponse({"data": result, "message": "SUCCESS"}, status=200)
+
+        return JsonResponse({"message": "PAGE_NOT_FOUND"}, status=404)
+
+        
+class ProductDetailView(View):
+    def get(self, request, product_id):
+        try:
+            product  = Product.objects.get(id=product_id)
+            category_name = product.category.menu.name
+            context  = {
+                'category'           : category_name,
+                'id'                 : product.id,
+                'productImageSrc'    : product.image_set.get(is_main=False).image_url,
+                'productCardImageSrc': product.image_set.get(is_main=True).image_url,
+                'healthGoalList'     : [goal.name for goal in product.goal.all()],
+                'title'              : product.name,
+                'subTitle'           : product.sub_name,
+                'description'        : product.description,
+                'nutritionLink'      : product.nutrition_url,
+                'allergyList'        : [allergy.name for allergy in product.allergy.all()],
+                'dietaryHabitList'   : [dietary_habit.name for dietary_habit in product.dietary_habit.all()],
+                }
+
+        # is_vegan, is_vegeterian
+            vegan_level = product.vegan_level_id
+            if vegan_level == 1:
+                is_vegan      = True
+                is_vegeterian = False
+            elif vegan_level == 2:
+                is_vegan      = False
+                is_vegeterian = True
+            else:
+                is_vegan      = False
+                is_vegeterian = False
+
+            context['isVegan']        = is_vegan
+            context['isVegetarian']   = is_vegeterian
+
+        # size, price
+            product_SSPs = ProductStock.objects.filter(product=product)  # SSP: siz e, stock, price
+            if category_name == 'vitamins':
+                price      = product_SSPs.first().price   
+                is_soldout = bool(product_SSPs.first().stock == 0)
+            else:
+                price      = {product_SSP.size : product_SSP.price for product_SSP in product_SSPs}
+                is_soldout = {product_SSP.size : bool(product_SSP.stock == 0) for product_SSP in product_SSPs}
+
+            context['productPrice']     = price
+            context['isSoldOut']        = is_soldout
+
+        # similar products
+            for goal_product in product.goal.all():
+                similar_product_list = [
+                    {
+                        'id'            : similar_product.id,
+                        'title'         : similar_product.name,
+                        'subTitle'      : similar_product.sub_name,
+                        'imageUrl'      : similar_product.image_set.get(is_main=True).image_url,
+                        'healthGoalList': [goal.name for goal in similar_product.goal.all()],
+
+                        }
```

Comment:
```suggestion
                similar_product_list = [
                    {
                        'id'            : similar_product.id,
                        'title'         : similar_product.name,
                        'subTitle'      : similar_product.sub_name,
                        'imageUrl'      : similar_product.image_set.get(is_main=True).image_url,
                        'healthGoalList': [goal.name for goal in similar_product.goal.all()]
                    }
```
필요없는 공백은 제거해주시고, 같은 레벨의 데이터를 감싸고 있는 괄호는 위치를 맞춰주시는 것이 좋습니다 👍

Reply:
네 수정하겠습니다